Trappist/src/Promact.Trappist.Repository/TestSettings/TestSettingsRepository.cs

```diff
@@ -45,10 +37,28 @@ public string UpdateTestSettings(int id, Test testObject)
         /// </summary>
         /// <param name="id">The parameter "id" is used to get the Settings of a Test by its Id</param>
         /// <returns>Settings Saved for the selected Test</returns>
-        public Test GetTestSettings(int id)
+        public async Task<Test> GetTestSettingsAsync(int id)
+        {
+            string currentDate = DateTime.Now.ToString("MM/dd/yyyy HH:mm");
+            var testSettings = await _dbContext.Test.FirstOrDefaultAsync(x => x.Id == id);
+            if (testSettings != null)
+            {
+                testSettings.StartDate = testSettings.StartDate == default(DateTime) ? Convert.ToDateTime(currentDate) : testSettings.StartDate;
```

Comment:
Not sure what are you doing here? The condition does not makes sense. Please write comment on lines where you are writing something complicated for others to understand.